openmdao/drivers/scipy_optimizer.py

```diff
@@ -449,20 +448,38 @@ def run(self):
                               f"than min allowed ({info['min_improve_pct']:.1f}%)."
                         issue_warning(msg, prefix=self.msginfo, category=DerivativesWarning)
 
+        if MPI:
```

Comment:
Rather than doing all of this, you can just check rank with `self._problem_comm.rank`.  That's what we do in DOE and GA drivers.